gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/SetStrategy.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.common.model.impl.properties.strategies;
+
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataInputViewStreamWrapper;
+import org.apache.flink.core.memory.DataOutputViewStreamWrapper;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Strategy class for handling {@code PropertyValue} operations with a value of the type
+ * {@code Set}.
+ */
+public class SetStrategy extends AbstractVariableSizedPropertyValueStrategy<Set<PropertyValue>> {
+
+  @Override
+  public Set<PropertyValue> read(DataInputView inputView, byte typeByte) throws IOException {
+    byte[] rawBytes = readVariableSizedData(inputView, typeByte);
+
+    PropertyValue item;
+    Set<PropertyValue> set = new HashSet<>();
+    DataInputViewStreamWrapper internalInputView = createInputView(rawBytes);
+
+    try {
+      while (internalInputView.available() > 0) {
+        item = new PropertyValue();
+        item.read(internalInputView);
+
+        set.add(item);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error reading PropertyValue", e);
```

Comment:
Same here ... why converting to RuntimeE ?

Reply:
see above